xwiki-commons-core/xwiki-commons-xml/src/main/java/org/htmlcleaner/TrimAttributeCleanerTransformations.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.htmlcleaner;
```

Comment:
hmm so this is not something meant to be extended. We definitely need a javadoc comment below to explain why we're doing this and what is the good future solution, and reference an htmlcleaner issue to follow up and when fixed explain to remove this class.

Reply:
We should also mention that this class is not meant to be public

Reply:
Definitely super weird to have such a package and it deserve a big comment.

Reply:
I'll add a comment but FTR the package was already existing in our own code for XWikiDOMSerializer.

Reply:
Doesn't make it right IMO :)

If we don't need this package for technical reason we should not use it (so from what I see it was wrong in case of XWikiDOMSerializer).

Reply:
> (so from what I see it was wrong in case of XWikiDOMSerializer).

It was necessary ofc, or it wouldn't have been used... It's because we needed to call package protected methods.

Reply:
> we needed to call package protected methods

Does not seems to be needed anymore (we don't use the method anymore and it's now public anyway).

Reply:
Then it should be moved indeed.

Reply:
It's still needed for createSubNodes call in XWikiDOMSerializer and it's definitely needed for the new files I added.

Reply:
> it's definitely needed for the new files I added.

Should be documented then.

Reply:
Just done.